public/apps/configuration/panels/role-edit.tsx

```diff
@@ -47,24 +58,191 @@ const TITLE_TEXT_DICT = {
+) {
+  const panels = indexPermissions.map((perm, index) => {
+    const onComboBoxChangeHandler = (field: string) =>
+      updateElementInArrayHandler(setRoleIndexPermission, [index, field]);
+
+    const onCreateOptionHandler = (field: string) =>
+      appendOptionToComboBoxHandler(setRoleIndexPermission, [index, field]);
+
+    return (
+      <Fragment key={`index-permission-${index}`}>
+        <EuiAccordion
+          id={`index-permission-${index}`}
+          buttonContent={
+            perm.indexPatterns.map(comboBoxOptionToString).join(', ') || 'Add index permission'
+          }
+          extraAction={
+            <EuiButton
+              color="danger"
+              onClick={() => removeElementFromArray(setRoleIndexPermission, [], index)}
+            >
+              Remove
+            </EuiButton>
+          }
+        >
+          <FormRow headerText="Index" helpText="Specify index pattern using *">
+            <EuiComboBox
+              noSuggestions
+              placeholder="Search for index name or type in index pattern"
+              selectedOptions={perm.indexPatterns}
+              onChange={onComboBoxChangeHandler('indexPatterns')}
+              onCreateOption={onCreateOptionHandler('indexPatterns')}
+            />
+          </FormRow>
+          <FormRow
+            headerText="Index Permissions"
+            headerSubText="Specify permissions using either action groups or single permissions. An action group is a list of single permissions.
+        You can often achieve your desired security posture using some combination of the default permission groups. You can
+        also create your own reusable permission groups."
+          >
+            <EuiFlexGroup>
+              <EuiFlexItem style={{ maxWidth: '400px' }}>
+                <EuiComboBox
+                  placeholder="Search for action group name or permission name"
+                  options={permisionOptionsSet}
+                  selectedOptions={perm.allowedActions}
+                  onChange={onComboBoxChangeHandler('allowedActions')}
+                />
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiButton>Browse and select</EuiButton>
+              </EuiFlexItem>
+              <EuiFlexItem grow={false}>
+                <EuiButton iconType="popout" iconSide="right">
+                  Create Action Groups
+                </EuiButton>
+              </EuiFlexItem>
+            </EuiFlexGroup>
+          </FormRow>
+          <FormRow
+            headerText="Document level security"
+            headerSubText="You can restrict a role to a subset of document in an index."
+            helpLink="/"
+            optional
+          >
+            <EuiTextArea
+              placeholder={FIELD_LEVEL_SECURITY_PLACEHOLDER}
+              value={perm.dls}
+              onChange={e => onComboBoxChangeHandler('dls')(e.target.value)}
```

Comment:
is it intended to use onComboBoxChangeHandler in a EuiTextArea?

Reply:
Good point. Renamed `onComboBoxChangeHandler` to `onValueChangeHandler` to avoid confusing.